statistics/cmsketch.go

```diff
@@ -298,6 +298,28 @@ func (c *CMSketch) MergeCMSketch(rc *CMSketch) error {
 	return nil
 }
 
+// MergeCMSketch4IncrementalAnalyze merges two CM Sketch for incremental analyze.
+// Since there is no value that appears partially in `c` and `rc`, we can just
+// merge them using `max` operations.
+func (c *CMSketch) MergeCMSketch4IncrementalAnalyze(rc *CMSketch) error {
```

Comment:
Could you explain more on the rationale behind this? We don't remove upper bound from old CMSketch now, so there can be value that appears partially in `c` and `rc` now? Even if there is no value that appears partially in `c` and `rc`, why we can use `max` operation to merge them?

Reply:
When we query from the cm sketch, we use the `min` to get the answer

- For values that only appears in `c`, using `max` to merge them affects the `min` query result less than using `sum`.
- For values that only appears in `rc`, it is the same;
- For values that appears both in `c` and `rc`, if they do not appear partially in `c` and `rc`, for example, if the `v` appears 5 times in the table, it can appears `3` in `c` and `5` in `rc`, then `max` is also gives the correct answer.

So in fact, if we can know the number of appearances of each value in the first place, it is better to use `max` to construct the cm sketch rather than `sum`.

Reply:
from the first intuition, it should be the `sum` of these two CM-Sketches?

Reply:
It would be wrong in the third case.

Reply:
Got it, please add this explanation into the comment also.